src/Microsoft.OData.Client/Serialization/Serializer.cs

```diff
@@ -405,6 +610,132 @@ internal void WriteNestedResourceInfo(EntityDescriptor entityDescriptor, IEnumer
             }
         }
 
+        /// <summary>
+        /// Writes a nested resource
+        /// </summary>
+        /// <param name="entityDescriptor">The entity</param>
+        /// <param name="relatedLinks">The links related to the entity</param>
+        /// <param name="odataWriter">The ODataWriter used to write the navigation link.</param>
+        internal void WriteDeltaNestedResourceInfo(EntityDescriptor entityDescriptor, IEnumerable<Descriptor> relatedDescriptors, ODataWriterWrapper odataWriter)
+        {
+            Dictionary<string, List<Descriptor>> groupRelatedLinks = new Dictionary<string, List<Descriptor>>(EqualityComparer<string>.Default);
+            foreach (Descriptor end in relatedDescriptors)

```

Comment:
Since this is internal, we could keep the `IEnumerable<Descriptor>` as `List<Descriptor>` to make the `foreach` allocation-free.